components/sup/src/ctl_gateway/server.rs

```diff
@@ -303,7 +305,7 @@ impl SrvHandler {
                 Ok(CtlCommand::new(ctl_sender,
                                    msg.transaction(),
                                    move |state, req, _action_sender| {
-                                       commands::service_load(state, req, &m)
+                                       commands::service_load(state, req, m)
```

Comment:
Seeing this change got me thinking:

Given the changes to the `CtlCommand` future above, an in the interest of keeping changes as small and restricted as possible, I'd like to propose the following:

Convert this `m` to `m.clone()`, as is done in the other calls in this match statement, and then withhold the other changes to `CtlCommand`. 

It's not that I doubt the changes will work... it's just that they are a bit gnarly, and it would be nice to separate those from the overall change to `merge_svc_load`. The concrete usage of futures in this particular module has never quite "clicked" with me -- it has always felt like there's one-layer-too-many (might just be a sign I need to dig further, though) -- but I suspect there are bigger changes we'll want to make in how things are structured here. As such, I think it would be useful to keep changes like this to a minimum until we can take a more holistic view of how this whole subsystem is structured.

What do you think?

Reply:
Sounds completely reasonable. I have the same feeling. It seems like there should be a single layer where we switch to async calls whereas now they seem interspersed throughout. I am hoping that when we migrate to `std::future::Future` and async/await a natural dividing line for async operations emerges.

Is there an issue for migrating the async code? I know it is not stabilized yet, but I could open an epic and add an issue that this `Fn` can be converted to `FnBox`. That way we can track it.

Reply:
There is not an issue for migrating to standard library async code yet. If you could open an epic and add that `Fn`/`FnBox` issue, that'd be fantastic, thanks!